socorro/middleware/search_common.py

```diff
@@ -273,6 +273,9 @@ def fix_date_parameter(self, parameters):
                 if not param.operator:
                     # dates can't be a specific date
                     raise BadArgumentError('date must have a prefix operator')
+                today = datetimeutil.utc_now()
+                if param.value > today:
+                    raise BadArgumentError('future date')
```

Comment:
This, and the previous ones as well, is a bad use of the ``BadArgumentError`` class: the first argument must be the field name (here, ``date``). By default the error will say ``Bad value for parameter(s) 'field_name'``. If you want to specify a different error message, use ``msg='my error message'``. So something like this: 

``raise BadArgumentError('date', msg='date most not be in the future')``

(Note that you can fix the previous use, on line 275, as well: ``raise BadArgumentError('date', 'date must have a prefix operator')``)